### ISSUE TITLE
Docs: Getting Started - invalid eslint.config.mjs example

### ISSUE THREAD

**Issue by hidarikani** [typescript-eslint/typescript-eslint#10758]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [x] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

At the moment `eslint.config.mjs` example on Getting Started page is as follows:

```js
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  eslint.configs.recommended,
  tseslint.configs.recommended,
);
```

actually running eslint with this config results in following error:

```
% npm run lint

> your-app-name@1.0.0 lint
> eslint src


Oops! Something went wrong! :(

ESLint: 9.19.0

TypeError: Unexpected array.
    at flatTraverse (/Users/*****/Documents/GitHub/*****/node_modules/@eslint/config-array/dist/cjs/index.cjs:442:12)
    at flatTraverse.next (<anonymous>)
    at normalize (/Users/*****/Documents/GitHub/*****/node_modules/@eslint/config-array/dist/cjs/index.cjs:462:19)
    at async FlatConfigArray.normalize (/Users/*****/Documents/GitHub/*****/node_modules/@eslint/config-array/dist/cjs/index.cjs:928:30)
    at async ConfigLoader.calculateConfigArray (/Users/*****/Documents/GitHub/*****/node_modules/eslint/lib/config/config-loader.js:557:9)
    at async #calculateConfigArray (/Users/*****/Documents/GitHub/*****/node_modules/eslint/lib/config/config-loader.js:633:29)
    at async entryFilter (/Users/*****/Documents/GitHub/*****/node_modules/eslint/lib/eslint/eslint-helpers.js:281:33)
    at async NodeHfs.<anonymous> (file:///Users/*****/Documents/GitHub/*****/node_modules/@humanfs/core/src/hfs.js:574:24)
    at async NodeHfs.walk (file:///Users/*****/Documents/GitHub/*****/node_modules/@humanfs/core/src/hfs.js:614:3)
    at async globSearch (/Users/*****/Documents/GitHub/*****/node_modules/eslint/lib/eslint/eslint-helpers.js:323:26)
```

Package versions that I have installed:
```
"@eslint/js": "^9.19.0",
"eslint": "^9.19.0",
"typescript": "^5.7.2",
"typescript-eslint": "^8.22.0",
```

I was able to resolve the error by spreading typescript-eslint config:

```js
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  eslint.configs.recommended,
  ...tseslint.configs.recommended,
);
```

### Affected URL(s)

https://typescript-eslint.io/getting-started/

### Additional Info

By the way, on ESLint's [Getting Started](https://eslint.org/docs/latest/use/getting-started#manual-set-up) page. It's recommended to run setup wizard:

```sh
npm init @eslint/config@latest
```

The config created by the wizard also uses the spread: `...tseslint.configs.recommended`.

**Comment by JoshuaKGoldberg** [comment]
Works on my machine 🙂: https://github.com/JoshuaKGoldberg/repros/tree/tseslint-getting-started-no-spread

Are you sure the version of `typescript-eslint` in your `node_modules/` is the one mentioned in this issue, `8.22.0`? 

For reference, the `...` was made unnecessary in #10333.

**Comment by hidarikani** [comment]
First of all, thank you for taking the time to create a reproduction branch. I did try it and it works according to the documentation. I've realised I don't use `tseslint.config()` helper  (will update the issue accordingly), that's why I had to spread. And it [does match the docs](https://typescript-eslint.io/packages/typescript-eslint#config).

With helper

```js
export default tseslint.config(
  eslint.configs.recommended,
  tseslint.configs.recommended,
);
```

Without helper:

```js
export default [
  eslint.configs.recommended,
  ...tseslint.configs.recommended,
];
```

It's quite interesting that a JS file can throw TS errors by using `tseslint.config()` 😮 .
